library/psa_crypto_core.h

```diff
@@ -111,6 +117,22 @@ static inline int psa_is_key_slot_locked(const psa_key_slot_t *slot)
     return slot->lock_count > 0;
 }
 
+/** Test whether a key slot is locked for writing.
+ *
+ * A key slot is locked for writing iff its lock counter is
+ * equal to SIZE_MAX.
+ * If multi-threading is enabled, the caller must hold the
+ * global key slot mutex.
+ *
+ * \param[in] slot  The key slot to test.
+ *
+ * \return 1 if the slot is locked for writing, 0 otherwise.
+ */
+static inline int psa_is_key_slot_locked_for_writing(const psa_key_slot_t *slot)
+{
+    return slot->lock_count == SIZE_MAX;
```

Comment:
Please define a symbolic constant `PSA_SLOT_STATE_IS_CREATING` or some such instead of using a magic value in multiple places.